### ISSUE TITLE
Error on https://game285222.konggames.com/games/jamboska/gift-clicker-mobile-edition/frame/d905e27e-3dff-4be4-a465-431e0ffcebee/

### ISSUE THREAD

**Issue by jexi27** [ruffle-rs/ruffle#11239]
# Error Info
Error name: Error
Error message: panicked at 'capacity overflow', library/alloc/src/raw_vec.rs:525:5
Error stack:
```
Error: panicked at 'capacity overflow', library/alloc/src/raw_vec.rs:525:5
    at W.n.wbg.__wbg_new_87297f22973157c8 (https://unpkg.com/@ruffle-rs/ruffle/core.ruffle.ee5716355123d2025211.js:1:50966)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::hafc7beb469c6cb04 (wasm://wasm/02c3637e:wasm-function[8620]:0x82e6b0)
    at std::panicking::rust_panic_with_hook::h7f7102b82d51338f (wasm://wasm/02c3637e:wasm-function[5613]:0x751612)
    at core::panicking::panic_fmt::he4489d678d6570d5 (wasm://wasm/02c3637e:wasm-function[8469]:0x829a1d)
    at alloc::raw_vec::capacity_overflow::h743526e95bf6b28b (wasm://wasm/02c3637e:wasm-function[8236]:0x8238d1)
    at <core::iter::adapters::flatten::FlatMap<I,U,F> as core::iter::traits::iterator::Iterator>::next::ha0792ffbf69c626d (wasm://wasm/02c3637e:wasm-function[2633]:0x59d8e3)
    at <ruffle_render_wgpu::context3d::WgpuContext3D as ruffle_render::backend::Context3D>::process_command::h54e9288cf0147e56 (wasm://wasm/02c3637e:wasm-function[531]:0xb148d)
    at ruffle_core::avm2::object::context3d_object::Context3DObject::copy_bitmap_to_texture::h19ac733f891021c5 (wasm://wasm/02c3637e:wasm-function[4661]:0x700453)
    at ruffle_core::avm2::globals::flash::display3D::textures::rectangle_texture::upload_from_bitmap_data::ha9949a70beb0757a (wasm://wasm/02c3637e:wasm-function[2689]:0x5ad8e3)
    at ruffle_core::avm2::function::Executable::exec::hba6a908a607c674f (wasm://wasm/02c3637e:wasm-function[799]:0x2871ad)
```
AVM2 stack:
```
    at flash.display3D.textures::RectangleTexture/flash::display3D::textures::RectangleTexture::uploadFromBitmapData()
    at starling.textures::ConcreteTexture/uploadBitmapData()
    at starling.textures::Texture$/fromBitmapData()
    at starling.textures::Texture$/fromBitmap()
    at starling.textures::Texture$/fromEmbeddedAsset()
    at srcStarling::GameDisplayStarling()
    at starling.core::Starling/initializeRoot()
    at starling.core::Starling/initialize()
    at starling.core::Starling/onContextCreated()
    at flash.events::EventDispatcher/flash::events::EventDispatcher::dispatchEvent()
    at flash.display::Stage3D/flash::display::Stage3D::requestContext3D_internal()
    at MethodInfo-1469()
```

# Player Info
Allows script access: true
Renderer: wgpu
Adapter Backend: Gl
Adapter Name: "ANGLE (Intel, Intel(R) UHD Graphics Direct3D11 vs_5_0 ps_5_0, D3D11)"
Adapter Device Type: IntegratedGpu
Adapter Driver Name: ""
Adapter Driver Info: ""
Enabled features: Features(TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | PUSH_CONSTANTS)
Available features: Features(TEXTURE_COMPRESSION_BC | CLEAR_TEXTURE | MULTIVIEW)
Current limits: Limits { max_texture_dimension_1d: 16384, max_texture_dimension_2d: 16384, max_texture_dimension_3d: 2048, max_texture_array_layers: 256, max_bind_groups: 4, max_bindings_per_bind_group: 640, max_dynamic_uniform_buffers_per_pipeline_layout: 8, max_dynamic_storage_buffers_per_pipeline_layout: 0, max_sampled_textures_per_shader_stage: 16, max_samplers_per_shader_stage: 16, max_storage_buffers_per_shader_stage: 0, max_storage_textures_per_shader_stage: 0, max_uniform_buffers_per_shader_stage: 11, max_uniform_buffer_binding_size: 16384, max_storage_buffer_binding_size: 0, max_vertex_buffers: 8, max_buffer_size: 268435456, max_vertex_attributes: 16, max_vertex_buffer_array_stride: 255, min_uniform_buffer_offset_alignment: 256, min_storage_buffer_offset_alignment: 256, max_inter_stage_shader_components: 60, max_compute_workgroup_storage_size: 0, max_compute_invocations_per_workgroup: 0, max_compute_workgroup_size_x: 0, max_compute_workgroup_size_y: 0, max_compute_workgroup_size_z: 0, max_compute_workgroups_per_dimension: 0, max_push_constant_size: 96 }
Surface quality: high
Surface samples: 4
Surface size: Extent3d { width: 550, height: 400, depth_or_array_layers: 1 }

# Page Info
Page URL: https://game285222.konggames.com/games/jamboska/gift-clicker-mobile-edition/frame/d905e27e-3dff-4be4-a465-431e0ffcebee/?kongregate_host=www.kongregate.com
SWF URL: https://game285222.konggames.com/gamez/0028/5222/live/GiftClicker.swf?kongregate_game_version=1503588262

# Browser Info
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36 Edg/113.0.1774.57
Platform: Win32
Has touch support: false

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-05-27
Channel: nightly
Built: 2023-05-27T00:23:49.907Z
Commit: e98230c988e361decb6f5a108a486fe40ab43931
Is extension: false

# Metadata
width: 500
height: 700
frameRate: 100
numFrames: 1
swfVersion: 35
backgroundColor: undefined
isActionScript3: true
uncompressedLength: 7167407

**Comment by n0samu** [comment]
The same problem is happening in a lot of Starling games - I pointed it out here https://github.com/ruffle-rs/ruffle/pull/11027#issuecomment-1556969305

**Comment by Lord-McSweeney** [comment]
Fixed by #11245.
